pkg/kratos/handlers.go

```diff
@@ -132,7 +133,16 @@ func (a *API) handleCreateFlow(w http.ResponseWriter, r *http.Request) {
 			a.cookieManager.ClearStateCookie(w)
 		}
 	} else {
-		response, cookies, err = a.handleCreateFlowNewSession(r, aal, returnTo, loginChallenge, refresh)
+		responseFlow, cookies, err = a.handleCreateFlowNewSession(r, aal, returnTo, loginChallenge, refresh)
+		setCookies(w, cookies)
+
+		if r.Header.Get("Accept") != "application/json, text/plain, */*" {
```

Comment:
nit: add explanation

```suggestion
        // If the browser was redirected here, we can't return a json object
        // so we redirect the user to the login page with the flow id appended in 
        // the query params
		if r.Header.Get("Accept") != "application/json, text/plain, */*" {
```